public/app/features/explore/QueryRow.tsx

```diff
@@ -70,7 +41,7 @@ export class QueryRow extends PureComponent<QueryRowProps, QueryRowState> {
   onChange = (query: DataQuery, override?: boolean) => {
     const { datasourceInstance, exploreId, index } = this.props;
     this.props.changeQuery(exploreId, query, index, override);
-    if (query && !override && datasourceInstance.getHighlighterExpression && index === 0) {
+    if (query && !override && datasourceInstance?.getHighlighterExpression && index === 0) {
```

Comment:
Nothing wrong here i'm just curious, why is `datasourceInstance` optional?

Reply:
Because that's how it is typed in`public/app/types/explore.ts`.
![image](https://user-images.githubusercontent.com/30407135/125074817-12dfd880-e0be-11eb-8569-2e7308de7ea6.png)

Reply:
So when we get it from store, it is going to have the same type.

Reply:
Oh i see, that makes sense! 👍

Reply:
Previously, when we didn't use ConnectedProps, we could type it (incorrectly) by ourself. And we typed it as DataSourceAPI, which wasn't correct. So when I refactored to use ConnectedProps, I discovered this 🐛 and add handling for cases, when datasourceInstance is undefined/null.
![image](https://user-images.githubusercontent.com/30407135/125075132-81bd3180-e0be-11eb-901e-5d4485584a4e.png)

Reply:
Oh that's nice, good job 🌟